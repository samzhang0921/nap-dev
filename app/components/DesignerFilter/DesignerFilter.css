.scrollPanel {
  overflow: hidden;
  padding: 0px;
  outline: none;
  width: 183px;
  max-height: 283px;
  outline: 0;
  clear: both;
  display: block;
  height: auto;
  position: relative;
}
