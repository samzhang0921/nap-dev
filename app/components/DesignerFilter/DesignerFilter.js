import React from 'react';
import { render } from 'react-dom';
import Designer from '../Designer/Designer';
export default class DesignerFilter extends React.Component{
  constructor(props) {
    super(props)
  }

  render() {
    if(!this.props.designerList) {return <div>Loading</div>};
    const designerList = this.props.designerList;
    return (
      <div>
        <ul>
          {designerList.map((designer) => {
            console.log('designer', designer)
            return <Designer key = {designer.id} designerName = {designer.name.en} designerId = {designer.id}/>
          })}
        </ul>
      </div>
    )
  }

}
