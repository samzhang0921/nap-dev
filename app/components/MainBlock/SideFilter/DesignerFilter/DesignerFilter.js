import React from 'react';
import { render } from 'react-dom';
import style from './DesignerFilter.css';
import Designer from '../Designer/Designer';
import PropTypes from 'prop-types';

class DesignerFilter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      clearClicked:false
    };
    this.clearDesigner = this.clearDesigner.bind(this);
  }

  clearDesigner () {
    const {clearClicked} = this.state;
    this.setState({
        clearClicked:!clearClicked
    });
    this.props.updateDesignerList(0);
  }
  componentWillUpdate(nextProps, nextState){
    if (this.props.categoryRefresh !== nextProps.categoryRefresh){
      this.setState({
        clearClicked:!this.state.clearClicked
      })
    }
  }

  render() {
    if(!this.props.designerList) {return <div className={style.DesignerFilter} >Loading</div>};
    const designerList = this.props.designerList;
    return (
      <div className={style.DesignerFilter}>
        <h4>Designer</h4>
        <a className={style.clearFilter} onClick={()=>this.clearDesigner()}>Clear</a>
        <ul className={style.scrollPanel}>
            {designerList.map((designer) => {
              return <Designer
              key = {designer.id}
              designerName = {designer.name.en}
              designerId = {designer.id}
              updateDesignerList = {this.props.updateDesignerList}
              clearClicked = {this.state.clearClicked} />
            })}
        </ul>
      </div>
    )
    }
};
DesignerFilter.propTypes = {
  designerList:PropTypes.array,
  updateDesignerList : PropTypes.func,
  categoryRefresh : PropTypes.bool
};
export default DesignerFilter;
