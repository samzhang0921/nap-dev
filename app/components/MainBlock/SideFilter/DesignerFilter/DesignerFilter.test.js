import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import DesignerFilter from './DesignerFilter';

it('should render a todo item ', () => {
  const tree = toJson(shallow(<DesignerFilter data={[]} />));
  expect(tree).toMatchSnapshot();
});

it('should match its snapshot with items', () => {
  const data = ['a', 'b', 'c'];
  const tree = renderer.create(
    <DesignerFilter data={data} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
