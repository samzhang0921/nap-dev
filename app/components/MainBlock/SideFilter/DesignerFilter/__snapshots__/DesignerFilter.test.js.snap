// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match its snapshot with items 1`] = `
<div
  className="DesignerFilter"
>
  Loading
</div>
`;

exports[`should render a todo item  1`] = `
<div
  className="DesignerFilter"
>
  Loading
</div>
`;
